### ISSUE TITLE
model.save() Error: object has no attribute '_compile_metrics'

### ISSUE THREAD

**Issue by dodiak** [tensorflow/tensorflow#36135]
<em>With python3 model saving stopped to work.</em>

After switching from python2 to python3 I started to get for my not compiled model the following error:
Error: 'Model' object has no attribute '_compile_metrics' 
Later I switched to TF 2 - right now running with 2.0.1 - but nothing has changed.

**System information**
- Ubuntu, Debian
- TensorFlow installed from binary
- TensorFlow version from 1.x -> 2.0.1
- Python version: 3.5.2
- CUDA/cuDNN version: 8 -> 10.1  based on TF needs
- GPU model and memory: Various.

**Comment by gadagashwini** [comment]
@dodiak, Can you provide the complete code to reproduce the reported issue. Thanks!

**Comment by dodiak** [comment]
@gadagashwini 
Thanks and sorry for the late reply. Some significant minimization was needed.
Here it is:
import os
import tensorflow as tf
import tensorflow.keras.backend as K
from tensorflow.keras import models
import numpy as np
os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2' # 0:default all | 1:no info | 2:no warning | 3:no error

x=np.ndarray(shape=(8,6),dtype=int,buffer=np.array([[0,2,3,4,5,1],[0,3,3,1,5,1],[0,2,5,4,5,1],[0,2,5,4,5,1],[0,2,3,4,5,1],[0,2,3,1,1,1],[0,2,1,4,5,1],[0,2,3,4,5,1]]))
y=np.ndarray(shape=(8,6,1),dtype=int,buffer=np.array([[[2],[3],[4],[5],[1],[1]],[[3],[3],[1],[5],[1],[1]],[[2],[3],[4],[5],[1],[1]],[[3],[3],[1],[5],[1],[1]],[[2],[5],[4],[5],[1],[1]],[[2],[3],[1],[1],[1],[1]],[[2],[1],[4],[5],[1],[1]],[[2],[3],[4],[5],[1],[1]]]))
tf.compat.v1.disable_eager_execution()

kwargs = dict()
graph = tf.Graph()
with graph.as_default():
  batch_size = tf.compat.v1.placeholder(tf.int64, shape=[])
  batch_x = tf.compat.v1.placeholder(tf.int64, shape=(None,None))
  batch_y = tf.compat.v1.placeholder(tf.int64, shape=(None,None,1))
  sample_weights = tf.compat.v1.placeholder(tf.float32, shape=(None,None))
  inputs = [batch_x, batch_y, sample_weights, batch_size]
  inputs += [K.learning_phase()]
  model_input = tf.keras.layers.Input(tensor=batch_x)
  optimizer = tf.keras.optimizers.Adagrad(**kwargs)
  lastOutput=tf.keras.layers.Embedding( 6, 300, name='embedding', embeddings_regularizer = None )(model_input)
  lastOutput=tf.compat.v1.keras.layers.CuDNNLSTM( 256, kernel_regularizer = None, recurrent_regularizer = None, bias_regularizer = None, return_sequences = True, stateful = False, **kwargs )(lastOutput)
  last_hidden = lastOutput
  model_output = tf.keras.layers.Dense( 6, activation = 'softmax', name='output_L', kernel_regularizer = None, bias_regularizer = None )(lastOutput)
  model = tf.keras.models.Model( inputs=model_input, outputs=model_output )
  # config model without compile
  model.optimizer = optimizer
  model.loss = 'sparse_categorical_crossentropy'
  #self.model.metrics = []
  labels = tf.reshape(batch_y, (-1,1))
  sample_weights = tf.reshape(sample_weights, (-1,1))
  last_hidden = tf.reshape(last_hidden, (-1,last_hidden.shape[-1]))
  w,b = model.layers[-1].weights
  logits = tf.matmul( last_hidden, w ) + b
  total_loss = tf.compat.v1.losses.sparse_softmax_cross_entropy( labels=tf.reshape(labels, (-1,)), logits=logits, weights=sample_weights )
  # batch training 
  with K.name_scope('training'):
    with K.name_scope(optimizer.__class__.__name__):
      training_updates = optimizer.get_updates(params=model.trainable_weights, loss=total_loss)
    updates = model.updates + training_updates
    model.train_function = K.function( inputs, [total_loss], updates = updates )
  model.predict_function = K.function( [batch_x, K.learning_phase()], [model_output], updates=model.state_updates )

  print("## Train Model ##")
  inputs = [x, y, y.shape[0]*y.shape[1], 1.]
  loss = model.train_function(inputs)[0]
  print("## Save Model ##")
  model.save('/tmp/model.hdf5')

**Comment by gadagashwini** [comment]
@dodiak, Tried replicating the reported issue but getting different error. Please take a look at [gist](https://colab.research.google.com/gist/gadagashwini/51289f56f062db7ca715e11ce5964596/untitled370.ipynb) and confirm. Thanks!

**Comment by dodiak** [comment]
ok, copy/paste error. I pasted at gist one more times the full code and it
give the same error as I see here.
thanks

**Comment by gadagashwini** [comment]
@dodiak, I don't see the gist in the comment. Can you include the gist in comment box. Thanks!

**Comment by dodiak** [comment]
Sorry, first time using gist, wo one more times:
https://colab.research.google.com/gist/dodiak/cf98f0eb5c0ecf7daad735ac05102f4d/untitled370.ipynb

**Comment by gowthamkpr** [comment]
@dodiak I have ran your code using tf-nightly and I am not running into any errors. Please find my gist [here](https://colab.sandbox.google.com/gist/gowthamkpr/4683486c67976a2accce02830a29685c/untitled370.ipynb#scrollTo=p3-KfWu3sUap). Thanks!

**Comment by dodiak** [comment]
Great, thanks a lot. I can confirm that with:
tf-nightly             2.2.0.dev20200218  
tf-nightly-gpu         2.2.0.dev20200227  
the simple example works. Give me a few more days please to test with a full version.

**Comment by gowthamkpr** [comment]
Sure. I am closing this issue now. If you have any more questions please open a new issue. Thanks!

**Comment by dodiak** [comment]
So I can confirm that it runs as we expected. Thanks a lot.
